### PR TITLE
Refactor statistics

### PR DESCRIPTION
**Pull request by magalilemes**
This is related to #276 and tries to improve the current structure of the `src/statistics`.sh file.

### CODE REVIEW

**Comment by alan-barzilay** [comment]
@magalilemes, the problem shellcheck found is that it cannot understand where configurations came from. If I'm not mistaken this is because configurations is defined elsewhere and then imported here with include, right? I think that by using the include function shellcheck may not be able to follow the imports and understand what is being imported. Maybe we should add this error to the the list of things to be ignored, or rethink (again) how we deal with imports (I think import guards wouldn't have this problem)

File: src/statistics.sh [hunk 1]
```diff
@@ -17,82 +17,83 @@ declare -gA shared_data=(["deploy"]='' ["build"]='' ["list"]='' ["uninstall"]=''
...
-  fi
-
-  case "$info_request" in
+  case "$target" in
     --day)
-      if [[ -z "$date_param" ]]; then
-        target_day="$KW_DATA_DIR/statistics/$year_month_dir/$day"
-      else
-        target_day="$KW_DATA_DIR/statistics/$(date_to_format "$date_param")"
+      local day
```
Comment (alan-barzilay) [inline src/statistics.sh:42]: @rodrigosiqueira, do we want to keep the `local` declarations all together in the beginning of the function? you mentioned something like this

Comment (rodrigosiqueira) [reply src/statistics.sh:42]: Yeah... let's do that since this can avoid shellcheck problems.

**Comment by rodrigosiqueira** [comment]
> @magalilemes, the problem shellcheck found is that it cannot understand where `configurations` came from. If I'm not mistaken this is because configurations is defined elsewhere and then imported here with include, right? I think that by using the include function shellcheck may not be able to follow the imports and understand what is being imported. This could also be seen as a counter part for the[ SC2034](https://github.com/koalaman/shellcheck/wiki/SC2034) error
> Maybe we should add this error to the the list of things to be ignored, or rethink (again) how we deal with imports (I think import guards wouldn't have this problem)

Btw, using `configurations` in the way that Magali did, is something common around kw and harmless in this specific case. So let's ignore this check.

**Comment by alan-barzilay** [comment]
> So let's ignore this check.

@rodrigosiqueira I have created PR #331 for this

File: src/statistics.sh [hunk 2]
```diff
@@ -291,11 +287,13 @@ function week_statistics()
 
 function month_statistics()
 {
-  local month_path="$KW_DATA_DIR/statistics/$1"
+  local month="$1"
+  local month_path="$KW_DATA_DIR/statistics/$month"
   local all_data=""
+  local pretty _month
```
Comment (joaoseckler) [inline src/statistics.sh:293]: I think there is a left over space!

**Review by joaoseckler — CHANGES_REQUESTED** [review]
Other than the one little change I commented, LGTM!

**Comment by rodrigosiqueira** [comment]
Merged, thanks a lot.

**Pull request closed without merging**